[docs]: Add wildcard and remove gomods documentation (#10)
txtdirect/website: content/docs/specification/content.md

## Wildcards

> Beside using wildcards to handle [non-existent zones](#wildcard-records), they can also get used to generate zones using a request's path. Using the `path` record you can use the incoming request's path to match a wildcard record and use its TXT record.

Reviewer: ```suggestion
If a specific record is non existent it will fallback to a wildcard record, if available. 
Wildcard records can be set using `_` such as `_redirect._.path.example.com`.

*For non existent zones or catch-all use-cases a [general wildcard](#wildcard-records) can be used.*
```

- Key: **from**
- Permitted values: "simplified regex"

> - Example: "from=/$2/$1/\$3"

Reviewer: *cough*

Author: Damn it. Not gonna use that extension anymore :man_facepalming:

- Mandatory
- Permitted values: "absolute URL"
- Example: "to=https://example.com/subpage/"

---

# Wildcard Records

A wildcard DNS record is a record in a DNS zone that will match requests for non-existent domain names. For example you can specifiy a record that matches all the subdomains that don't have a specific TXT record.

Take a look at these sample records:

```

> 

Reviewer: ```suggestion
; NOTE: Using a CNAME record for the wildcard is not supported and will lead to wrong behaviour
*.example.test.                    IN A   127.0.0.1
_redirect._.example.test.          IN TXT "v=txtv0;to=http://wildcard.test;type=host;code=302"

; NOTE: Specific records can use a CNAME or A/AAAA records.
specific.example.test              IN A   127.0.0.1
_redirect.specific.example.test.   IN TXT "v=txtv0;to=http://specific.test;type=host;code=302"
```

> example.test.                    IN A   127.0.0.1

Reviewer: ```suggestion
```

> _redirect.example.test.          IN TXT "v=txtv0;to=http://example.com;type=host;code=302"

Reviewer: ```suggestion
```

> _redirect.specific.example.test. IN TXT "v=txtv0;to=http://specific.test;type=host;code=302"

Reviewer: ```suggestion
```

> _redirect._.example.test.        IN TXT "v=txtv0;to=http://wildcard.test;type=host;code=302"

Reviewer: ```suggestion
```

> 

Reviewer: Whitespace?

Author: All the whitespaces in snippets are because of the extra space bug. Just thought it would look better if they all were aligned.

Reviewer: I understand the logic, but I feel this will be left, if the bug is fixed. Let's remove it and fix the bug in Syna faster. Or bug @mpourismaiel more :)

Author: Alright. Removed them and bugged Mahdi once again :smile:

```

> 

Reviewer: Whitespace?

> All the requests for a subdomain of `example.test` like `test.example.test` would get redirected to `http://wildcard.test` by the last record because they don't have a specific record. Expect for the `specific.example.test` subdomain which has a specific record for itself.

Reviewer: ```suggestion
All the requests for a given subdomain of `example.test` such as `test.example.test` would use the wildcard record of the parent zone `_redirect._.example.test`.

In the case a more specific record is available it will take be used.
For a specific record to work a specific subdomain needs to be setup as well. This is based on DNS specifics.
With `specific.example.test` and `_redirect.specific.example.test` set it will be used instead of the above wildcard.
```

The redirect flow for these records would look like this:

```

> 

Reviewer: Whitespace?

> example.test          -> http://example.com

Reviewer: ```suggestion
```

Wildcard + specific is already complicated. Adding this as well seems too much.
It is normal TXTDirect behaviour that one could add, but here it just adds more to think about I think.

specific.example.test -> http://specific.test
*.example.test        -> http://wildcard.test

> 

Reviewer: Whitespace?
